src/managers/config.py

```diff
@@ -591,24 +595,23 @@ def controller_properties(self) -> list[str]:
         Returns:
             List of properties to be set
         """
-        # TODO: this check will need to be done on relation data
-        roles = "process.roles="
+        if self.state.kraft_mode == False:  # noqa: E712
+            return []
+
+        roles = []
+        node_id = self.state.unit_broker.unit_id
+        if self.state.runs_broker:
+            roles.append("broker")
         if self.state.runs_controller:
-            roles += self.config.roles
-        # TODO node.id based on unit number for colocated, if only controller unit number + 100
-        # FIXME node.id will have to be a crunched form of model uuid + app name or similar
-        # FIXME quorum_voters into abstraction on state -> can be created from peer_orchestrator
-        # relation or from own cluster if controller runs colocated.
-        quorum_voters = [f"{broker.unit_id}@{broker.host}:{CONTROLLER_PORT}" for broker in self.state.brokers]
+            roles.append("controller")
+            node_id += 100
```

Comment:
**todo:** We should add/use literals for the node-offset, and for the role strings, I think.